### PR TITLE
SONARJAVA-1084 Remove usage of ASTNode in AnonymousClassesTooBigCheck

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by benzonico** [comment]
LGTM, at one point though the first/lastSyntaxTokenFinder should probably be method within the SyntaxTree. But this is not required right now, so let's live with this.

File: java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java [hunk 1]
```diff
@@ -187,17 +231,215 @@ public void visitVariable(VariableTree tree) {
...
+      lastSyntaxToken = tree.closeParenToken();
+    } else {
+      scan(tree.identifier());
+    }
+  }
+
+  @Override
+  public void visitNewArray(NewArrayTree tree) {
+    if (!tree.initializers().isEmpty()) {
+      // FIXME should be the close brace
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java [hunk 2]
```diff
@@ -187,17 +231,215 @@ public void visitVariable(VariableTree tree) {
...
+  }
+
+  @Override
+  public void visitPrimitiveType(PrimitiveTypeTree tree) {
+    lastSyntaxToken = tree.keyword();
+  }
+
+  @Override
+  public void visitArrayType(ArrayTypeTree tree) {
+    // FIXME should be the last dimension
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java [hunk 3]
```diff
@@ -187,17 +231,215 @@ public void visitVariable(VariableTree tree) {
...
+  }
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    if (!tree.types().isEmpty()) {
+      scan(Iterables.getLast(tree.types()));
+    } else if (!tree.imports().isEmpty()) {
+      scan(Iterables.getLast(tree.imports()));
+    } else {
+      // TODO(SONARJAVA-547) Should be the semi-colon token
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java [hunk 4]
```diff
@@ -187,17 +231,215 @@ public void visitVariable(VariableTree tree) {
...
+      } else {
+        lastSyntaxToken = ((ModifierKeywordTree) lastModifier).keyword();
+      }
+    }
+  }
+
+  @Override
+  public void visitAnnotation(AnnotationTree annotationTree) {
+    if (!annotationTree.arguments().isEmpty()) {
+      // FIXME should be the close parenthesis
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/syntaxtoken/LastSyntaxTokenFinder.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-checks/src/main/java/org/sonar/java/syntaxtoken/FirstSyntaxTokenFinder.java [hunk 5]
```diff
@@ -294,4 +321,112 @@ public void visitNewClass(NewClassTree tree) {
...
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    scan(tree.leftOperand());
+  }
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    if (tree.packageName() != null) {
+      // TODO(SONARJAVA-547) Should be the package token
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/syntaxtoken/FirstSyntaxTokenFinder.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 new issues:
* 3 major
* 2 info

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 new issues:
* 3 major
* 2 info

**Pull request merged**